tests/clv/models/test_modified_beta_geo.py

```diff
@@ -0,0 +1,599 @@
+        cls.model = ModBetaGeoModel(cls.data)
+
+        # Also instantiate lifetimes model for comparison
+        cls.lifetimes_model = ModifiedBetaGeoFitter()
+        cls.lifetimes_model.params_ = {
+            "a": cls.a_true,
+            "b": cls.b_true,
+            "alpha": cls.alpha_true,
+            "r": cls.r_true,
+        }
+
+        # Mock an idata object for tests requiring a fitted model
+        cls.N = len(cls.customer_id)
+        cls.chains = 2
+        cls.draws = 50
+
+        mock_fit = create_mock_fit(
+            {
+                "a": cls.a_true,
+                "b": cls.b_true,
+                "alpha": cls.alpha_true,
+                "r": cls.r_true,
+            }
+        )
+
+        mock_fit(cls.model, cls.chains, cls.draws, cls.rng)
+
+    @pytest.fixture(scope="class")
+    def model_config(self):
+        return {
+            "a_prior": Prior("HalfNormal"),
+            "b_prior": Prior("HalfStudentT", nu=4),
+            "alpha_prior": Prior("HalfCauchy", beta=2),
+            "r_prior": Prior("Gamma", alpha=1, beta=1),
+        }
+
+    @pytest.fixture(scope="class")
+    def default_model_config(self):
+        return {
+            "a_prior": Prior("HalfFlat"),
+            "b_prior": Prior("HalfFlat"),
+            "alpha_prior": Prior("HalfFlat"),
+            "r_prior": Prior("HalfFlat"),
+        }
+
+    def test_model(self, model_config, default_model_config):
+        for config in (model_config, default_model_config):
+            model = ModBetaGeoModel(
+                data=self.data,
+                model_config=config,
+            )
+            model.build_model()
+            assert isinstance(
+                model.model["a"].owner.op,
+                pm.HalfFlat
+                if config["a_prior"].distribution == "HalfFlat"
+                else config["a_prior"].pymc_distribution,
+            )
+            assert isinstance(
+                model.model["b"].owner.op,
+                pm.HalfFlat
+                if config["b_prior"].distribution == "HalfFlat"
+                else config["b_prior"].pymc_distribution,
+            )
+            assert isinstance(
+                model.model["alpha"].owner.op,
+                pm.HalfFlat
+                if config["alpha_prior"].distribution == "HalfFlat"
+                else config["alpha_prior"].pymc_distribution,
+            )
+            assert isinstance(
+                model.model["r"].owner.op,
+                pm.HalfFlat
+                if config["r_prior"].distribution == "HalfFlat"
+                else config["r_prior"].pymc_distribution,
+            )
+            assert model.model.eval_rv_shapes() == {
+                "a": (),
+                "a_log__": (),
+                "b": (),
+                "b_log__": (),
+                "alpha": (),
+                "alpha_log__": (),
+                "r": (),
+                "r_log__": (),
+            }
+
+    def test_missing_cols(self):
```

Comment:
Can you use pytest mark parametrize to make these each a test case

Reply:
Done. 

Note that equivalent tests in `TestBetaGeoModel` are not parametrize. 
Doubting if we should do it as part of this PR, I'd rather address this in a separate PR